app/components/SheetMusic/comments/Comments.jsx

```diff
@@ -0,0 +1,50 @@
+import fluxMixin from 'flummox/mixin';
+import React from 'react';
+
+import Comment from './Comment';
+import ReplyBox from './ReplyBox';
+
+function retrieveInitialData(flux, id) {
+  let sheetMusicActions = flux.getActions('sheetmusic');
+  return Promise.all([
+    sheetMusicActions.getComments(parseInt(id, 10), flux),
+  ]);
+}
+
+export default React.createClass({
+  propTypes: {
+    /**
+     * The id of the sheet music.
+     */
+    id: React.PropTypes.number.isRequired,
+  },
+
+  mixins: [
+    fluxMixin({
+      comments: store => ({
+        comments: store.state.comments,
+      }),
+    }),
+  ],
+
+  componentDidMount() {
+    retrieveInitialData(this.flux, this.props.id);
```

Comment:
We should use `routeWillRun` so that `retrieveInitialData` is called on the server side. Being able to make API calls on the server-side is one of the many benefits of isomorphic rendering. Currently in this class, `retrieveInitialData` only being called on the client side.

https://github.com/pianoshelf/webclient-react/blob/master/app/components/Search/Browse.jsx#L150

```js
  statics: {
    routeWillRun({ flux, state }) {
      return retrieveInitialData(flux, state.location.query);
    },
  }
```

There should also be a check to make sure if `retrieveInitialData` is called on the server-side, it's not called again on the client-side. Like here: https://github.com/pianoshelf/webclient-react/blob/master/app/components/Search/Browse.jsx#L160

It's really weird and complicated. After we're done building the site we should really switch from Flummox to something like [Redux](http://redux.js.org/) for state management.